File: chaeryeon823/PGS_12924.js
// 숫자의 표현
// lv 2
// 수학, 구현

function solution(n) {
  let answer = 1;
  for (let i = 1; i < n; i++) {
    let sum = 0;
    let tmp = i;
    while (sum < n) {
      sum += tmp;
      tmp += 1;
    }
    if (sum === n) {
      answer++;
    }
  }
  return answer;
}

File: ahma0/PGS_12924.java
class Solution {

    public int search(int n) {
        int answer = 0;

        for (int i = 1; i * (i + 1) / 2 <= n; i++) {
            if ((n - (i * (i - 1) / 2)) % i == 0) {
                answer++;
            }
        }

        return answer;
    }

    public static void main(String[] args) {
        //        Solution s = new Solution();
        //        int n = s.search(new int[]{-1, 0, 3, 5, 9, 12}, 9);
        //        System.out.println(n);
    }
}
